app/assets/javascripts/discourse/app/components/admin-post-menu.gjs

```diff
@@ -0,0 +1,231 @@
+import Component from "@glimmer/component";
+import { inject as service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import { fn } from "@ember/helper";
+import and from "truth-helpers/helpers/and";
+import or from "truth-helpers/helpers/or";
+import not from "truth-helpers/helpers/not";
+import { action } from "@ember/object";
+import concatClass from "discourse/helpers/concat-class";
+
+export default class AdminPostMenu extends Component {
+  @service currentUser;
+  @service siteSettings;
+  @service store;
+  @service adminPostMenuButtons;
```

Comment:
> it feels like it should be a component

is the main smell for me. I am seeing components being spread out across services and `/components` and causing some confusion or unnecessary noise. But I am really nitpicking here, I think this use case is good!

Reply:
Sorry just unresolving because I think it is interesting too...currently we do things like this where we have a variable that is only accessible inside the module that we register stuff to via the plugin API:

https://github.com/discourse/discourse/blob/79863db5e6f88a525854305f9bdeafbd10070156/app/assets/javascripts/discourse/app/components/composer-editor.js#L60-L66

Is that another use case that we would use these kinds of services for? I do agree it would be much nicer to use a service for things like this, especially if we can make it more generic like DiscoursePluginRegistry :+1: